test/ffmpeg-vaapi/transcode/mpeg2.py

```diff
@@ -0,0 +1,66 @@
+##
+### Copyright (C) 2018-2019 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ..util import *
+from .transcoder import TranscoderTest
+
+spec = load_test_spec("mpeg2", "transcode")
+class to_avc(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "avc", "hwhw"))
+  @platform_tags(set(MPEG2_DECODE_PLATFORMS)&set(AVC_ENCODE_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      srcformat    = 'mpeg2',
+      dstformat    = 'avc',
+      dstextension = 'h264',
+      case         = case,
+      ttype        = 'hwhw',
+    )
+    self.transcode()
+
+class to_hevc(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "hevc", "hwhw"))
+  @platform_tags(set(MPEG2_DECODE_PLATFORMS)&set(HEVC_ENCODE_8BIT_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      srcformat    = 'mpeg2',
+      dstformat    = 'hevc',
+      dstextension = 'hevc',
+      case         = case,
+      ttype        = 'hwhw',
+    )
+    self.transcode()
+
+class to_mjpeg(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "mjpeg", "hwhw"))
+  @platform_tags(set(MPEG2_DECODE_PLATFORMS)&set(JPEG_ENCODE_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      srcformat    = 'mpeg2',
+      dstformat    = 'mjpeg',
+      dstextension = 'mjpeg',
+      case         = case,
+      ttype        = 'hwhw',
+    )
+    self.transcode()
+
+class to_mpeg2(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "mpeg2", "hwhw"))
+  @platform_tags(set(MPEG2_DECODE_PLATFORMS)&set(MPEG2_ENCODE_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      srcformat    = 'mpeg2',
```

Comment:
Is `srcformat` really needed?  It seems redundant and already implied in test suite design and case source files.

Reply:
using srcformat as part of reference yuv name for better naming convention

Reply:
If you delegate the format specifiers to the user config, then you don't need this.

Reply:
removed srcformat and dstformat